Source/WTF/wtf/text/CString.cpp

```diff
@@ -169,4 +169,30 @@ bool CStringHash::equal(const CString& a, const CString& b)
     return a == b;
 }
 
+enum class ASCIICase { Lower, Upper };
+
+template<ASCIICase type>
+CString convertASCIICase(std::span<const char8_t> input)
+{
+    if (input.empty())
+        return CString();
```

Comment:
This produces a null string when the input is a non-null empty string. In the past we have preferred to instead map null to null and empty to empty. I suggest we match the behavior of the `CString` constructors that takes a `std::span`. Also, I suggest we add unit tests to cover this behavior, checking for empty vs. null.

Reply:
I didn't know if you mean just adding the tests to CString itself or to the conversion functions, but considering that CString already has such tests, I'm adding them to the conversion functions.